docs(config): document precedence order of config files (#13447)
renovatebot/renovate: docs/usage/configuration-options.md

You can store your Renovate configuration file in one of the following locations:

renovate.json
renovate.json5
.github/renovate.json
.github/renovate.json5
.gitlab/renovate.json
.gitlab/renovate.json5
.renovaterc

> .renovaterc.json

Reviewer: Restore the list change. For `package.json` we can add a note that it's deprecated

You can store your Renovate configuration file in one of the following locations:

1. `renovate.json`
1. `renovate.json5`
1. `.github/renovate.json`
1. `.github/renovate.json5`
1. `.gitlab/renovate.json`
1. `.gitlab/renovate.json5`
1. `.renovaterc`
1. `.renovaterc.json`

> 1. `package.json` _(within a `"renovate"` section)_

Reviewer 2: Can one of the maintainers check the order of the files? Some files have moved around in the list.

Reviewer 3: https://github.com/renovatebot/renovate/blob/0b5ee0b40083644445cd5b42696d3f7eff94760a/lib/config/app-strings.ts#L1-L11

Thats the order

Reviewer 2: Good! Thanks for checking! 😉

Reviewer 2: This comment chain needs to be marked "resolved" by one of you with `write` rights before this will automerge properly. 😉

You can store your Renovate configuration file in one of the following locations:

> - `renovate.json`

Reviewer 2: > and the lack of formatting for this list items makes it much harder to read.

How about changing it into a numbered list? For example:

```markdown
1. `renovate.json`
1. `renovate.json5`
1. `.github/renovate.json`
```

Which looks like this:

1. `renovate.json`
1. `renovate.json5`
1. `.github/renovate.json`

> - renovate.json
> - renovate.json5
> - .github/renovate.json
> - .github/renovate.json5
> - .gitlab/renovate.json
> - .gitlab/renovate.json5
> - .renovaterc
> - .renovaterc.json

Reviewer 2: ```suggestion
- `renovate.json`
- `renovate.json5`
- `.github/renovate.json`
- `.github/renovate.json5`
- `.gitlab/renovate.json`
- `.gitlab/renovate.json5`
- `.renovaterc`
- `.renovaterc.json`
```

I'd like to use `monospaced` font for the file names. 😉

- `package.json` _(within a `"renovate"` section)_

> _Note_: Storing Renovate config in `package.json` is now deprecated and its support may be removed in the future.

Reviewer 2: ```suggestion
_Note_: Storing the Renovate configuration in a `package.json` file is deprecated and support may be removed in the future.
```

> When renovating a repository, Renovate will try to autodetect/touch those files in exactly the same order they are listead above.

Reviewer 2: ```suggestion
When renovating a repository, Renovate will try to detect the configuration files in the order listed above.
```

How about this sentence?

Does Renovate `touch`/edit/change its own configuration files? I think that is something that only the end-user does?

Reviewer: Renovate does not touch them, this sentence was incorrect
